### ISSUE TITLE
Shipyards dont work on child-entities

### ISSUE THREAD

**Issue by CoasterBlock** [tsunamayo/Starship-EVO#5202]
### Short Description of the issue:

Shipyards dont work if they are built onto the station/mothership using a child-entity. 

### Steps to reproduce the issue:

Build a ship/station
add a hinge or piston or rotor
place a shipyard onto the child-entity

Shipyard is not selectable, and isnt detected by other ships. It is however visible and dockable to the ship on which the child-entity is built. So basically the ship would be able to dock to itself.

### Environment
- Starship EVO version: 23W30A
- Graphics Card: RTX 2080 Super
[Player.log](https://github.com/tsunamayo/Starship-EVO/files/12198131/Player.log)

### Additional information:
I included the design I used to test this potential bug as well. Maybe it has something to do with that. 

[Landing Pad 2.zip](https://github.com/tsunamayo/Starship-EVO/files/12198137/Landing.Pad.2.zip)

**Comment by TIKIRobo** [comment]
I have also had a problem similar to this where I am unable to target my own ships and so cannot select the "request landing" option, i am still able to target things like gas clouds and the habitat though, just not my ships

**Comment by CoasterBlock** [comment]
@TIKIRobo Yeah same, I was able to "target" things. But the autopilot doesnt work. It just shoots straight past everything. Habitats, stations, planets etc. It doesnt deactivate LC once a certain distance away
